File: app/src/main/java/com/minwook/mytriphistory/java/ui/activity/MainActivity.java
package com.minwook.mytriphistory.java.ui.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;

import com.google.android.material.tabs.TabLayout;
import com.minwook.mytriphistory.R;
import com.minwook.mytriphistory.java.adapter.MainTabAdapter;

public class MainActivity extends AppCompatActivity {
    private ViewPager viewPager;
    private TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tabLayout = findViewById(R.id.tabLayout);
        viewPager = findViewById(R.id.viewPager);

        MainTabAdapter mainTabAdapter = new MainTabAdapter(getSupportFragmentManager());
        viewPager.setAdapter(mainTabAdapter);

        tabLayout.setupWithViewPager(viewPager);
        viewPager.addOnPageChangeListener(new ViewPagerPageChangeListener());
        //viewPager.addOnAdapterChangeListener();
    }

    class ViewPagerPageChangeListener implements ViewPager.OnPageChangeListener {

        @Override
        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

        }

        @Override
        public void onPageSelected(int position) {

        }

        @Override
        public void onPageScrollStateChanged(int state) {

        }
    }
}

File: app/src/main/java/com/minwook/mytriphistory/kotlin/ui/activity/MainActivity.kt
package com.minwook.mytriphistory.kotlin.ui.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.minwook.mytriphistory.R
import com.minwook.mytriphistory.java.adapter.MainTabAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mainTabAdapter = MainTabAdapter(supportFragmentManager)
        viewPager.adapter = mainTabAdapter

        tabLayout.setupWithViewPager(viewPager)
    }
}